models/project/note.go

```diff
@@ -0,0 +1,370 @@
+// Copyright 2024 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package project
+
+import (
+	"context"
+	"fmt"
+
+	"code.gitea.io/gitea/models/db"
+	user_model "code.gitea.io/gitea/models/user"
+	"code.gitea.io/gitea/modules/markup/markdown"
+	"code.gitea.io/gitea/modules/setting"
+	"code.gitea.io/gitea/modules/timeutil"
+	"code.gitea.io/gitea/modules/util"
+
+	"xorm.io/builder"
+)
+
+// BoardNote is used to represent a note on a boards
+type BoardNote struct {
```

Comment:
This will create a table called `board_note`. I suggest to change the table name to `project_board_note`, so it alligns with the other `project_...` tables. Have a look at the `TableName()` method in other files. Also you need a migration if you add a table.

Reply:
What do I need to do for a migration?

Is it better to rename the struct or only the the table_name with `TableName()`?